### PR TITLE
Adding a vectorized implementation of TensorPrimitives.Log

### PR DESCRIPTION
**Pull request by tannergooding**
This makes progress towards https://github.com/dotnet/runtime/issues/92219

### CODE REVIEW

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs [hunk 1]
```diff
@@ -2579,6 +2579,226 @@ public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y)
...
+                    specialResult,
+                    n * Vector256.Create(V_LN2) + q
+                );
+            }
+
+#if NET8_0_OR_GREATER
+            public static Vector512<float> Invoke(Vector512<float> x)
+            {
+            }
+#endif
```
Comment (stephentoub) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs:2799]: This doesn't look right :smile:

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs [hunk 2]
```diff
@@ -2579,6 +2579,291 @@ public static Vector512<float> Invoke(Vector512<float> x, Vector512<float> y)
 #endif
         }
 
+        private readonly struct LogOperator : IUnaryOperator
+        {
+            // This code is based on `vrs4_logf` from amd/aocl-libm-ose
```
Comment (stephentoub) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs:2584]: We already have the right callouts in the thirdparty file, right?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs (hunk 2 above)
Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs:2584]: Yep! We have the general encompassing license notice here: https://github.com/dotnet/runtime/blob/main/THIRD-PARTY-NOTICES.TXT#L1139-L1164

And then the explicit callout like this in each function that's ported something, so users have something to refer back to/cross reference more easily.

**Comment by stephentoub** [comment]
Failure is https://github.com/dotnet/runtime/issues/92944

**Pull request merged**